### DISCUSSION TITLE
3D printable case

### DISCUSSION THREAD

**Discussion by paulsijben** [probonopd/MiniDexed#165] [Show and tell]
Has anyone come up with a 3D printable case that mimics the styling of the original synths?

**Comment by craigyjp** [comment]
I built it into this one to make it more professional....![20220422_162503.jpg](https://user-images.githubusercontent.com/51026305/164723588-38e2e279-a6a3-4628-b167-a1540575b5b9.jpg)

**Reply by eLuSiVeMiTe** [reply]
Didn't think about racking it. Nice idea.  Easy enough to fold a case up.

**Comment by ebrown3k** [comment]
I am working on a 3D printed case in the form factor of a Korg Volca. The issue is it will take some special parts to get it to fit in the limited height complicating the build. I will also be custom designing it for the specific Pi version and DAC I ordered so I don't know how useful it would really be for a wide range of users.

**Comment by dcoredump** [comment]
My case is not ready yet, but here are some pictures.

![PXL_20220422_181629662.jpg](https://user-images.githubusercontent.com/5188678/164771930-e5ee5c73-7557-4eb7-9626-9b25919e4836.jpg)

![PXL_20220422_181637060.jpg](https://user-images.githubusercontent.com/5188678/164771900-826b86cf-445f-4916-af9d-8ec98a2b5e91.jpg)

The DIN-MIDI is currently missing and I will add a (passive) volume pot. After that I will paint it black (only the metal parts).

**Comment by probonopd** [comment]
For extra authenticity, I recommend a 3D printed knob in this color:

![image](https://user-images.githubusercontent.com/2480569/164776134-95d26d77-0aa8-4d3c-8e02-bf13832669dc.png)

**Reply by paulsijben** [reply]
number of knobs (for comparison) are being printed in that color at the moment ;-)

**Reply by paulsijben** [reply]
luckily I had some filament in this color at hand

**Comment by probonopd** [comment]
Working on a 3d printable case (here: for the Raspberry Pi Zero) that can be assembled without any tools.

We'd need a whole range of them to accomodoate for the various RPi and DAC models.

There is something about snap-fit 3d printed cases that I really like. Maybe it's their hackability which makes everything accessible at any time without any tools.

Turn on the sound to hear those satisfying clicks :100: 

https://user-images.githubusercontent.com/2480569/164972750-211dfeb2-eb15-4ba7-a175-af313d03468a.mp4

**Reply by sail2themoon** [reply]
Is there a release of this somewhere? I'd like to make a SSD1306-version

**Reply by probonopd** [reply]
I never released that one as it proved to be so tight inside that I accidentally shorted some pins on my RPi and I wanted to protect others from that disaster. Maybe I (or someone else) will design a new one some day - this one might give you some inspiration ;.)

**Comment by paulsijben** [comment]
> Working on a 3d printable case (here: for the Raspberry Pi Zero) that can be assembled without any tools.
> 
> We'd need a whole range of them to accomodoate for the various RPi and DAC models.
> 
> There is something about snap-fit 3d printed cases that I really like. Maybe it's their hackability which makes everything accessible at any time without any tools.
> 
> Turn on the sound to hear those satisfying clicks 💯
> 
>  3dprintedcase.mp4

nice! 
I was still in doubt whether to use the zero 2 or the pi 3 leveraging the latter's USB ports

**Comment by probonopd** [comment]
The connectors on the Zero suck. A Pi 3 Model A might be the sweet spot, unless we start using Networking one day.

**Reply by rcarmo** [reply]
I'm actually using one and designing a case for it, but I want to inset the encoder so that I can have a disk-like knob that doesn't protrude from the case.

**Reply by probonopd** [reply]
Next time around I would also insert the screen at an angle, not like in my video.

**Reply by rcarmo** [reply]
Well, you can tune the contrast on the LCD a bit, and that usually changes the viewing angle... I think a 10KOhm pot is what's typical.

**Comment by paulsijben** [comment]
> The connectors on the Zero suck. A Pi 3 Model A might be the sweet spot, unless we start using Networking one day.

true but I have never seen a 3A in the flesh so to speak. I do have a 3b laying around ;-)

**Comment by probonopd** [comment]
After a weekend of modeling, lots of trial and error improving my snap-fit design rules for 3D printing: It's alive! (Not the final colors though.)

Everything fits barely in, including a DAC. No screws, no tools for assembly. (Besides a soldering iron.)

![minidexed-1](https://user-images.githubusercontent.com/2480569/164996673-5de60558-3bf9-416d-8545-f05bc6e7b3fb.jpg)

**Reply by JoaquinSainz** [reply]
Wooww nice!!

**Comment by craigyjp** [comment]
Nicely done

**Comment by eLuSiVeMiTe** [comment]
Only a temp case. Just need to make a new side panel to fit the screen/encoder.
![IMG_20220424_231356_679](https://user-images.githubusercontent.com/103902615/165000993-2824ac1e-eb37-41f8-8c6f-2adadf0db7cc.jpg)
![IMG_20220424_231429_623](https://user-images.githubusercontent.com/103902615/165000996-79f5c541-8882-4e70-a6e8-806f7b646acb.jpg)

**Reply by probonopd** [reply]
Welcome @[eLuSiVeMiTe](https://github.com/eLuSiVeMiTe).
Something didn't work with your upload, please upload again...

**Reply by eLuSiVeMiTe** [reply]
Should be there now

**Reply by probonopd** [reply]
Nice. You could make a 3D printed front panel simulating the 8 modules of a TX816 under your display...

![image](https://user-images.githubusercontent.com/2480569/165001504-80d8ca5b-4605-42bc-9c01-19abbeb4faed.png)

**Comment by now-its-dark** [comment]
Hey there, I've been sketching out a few case directions, trying to remain loyal to the design choices from Yamaha products of the era.  First one is in-progress, inspired by my personal favorite & most starkly 80's — the TQ5. 

Thinking about doing one inspired by the QX3 sequencer next. It would be awesome if there were some way to use mechanical keyboard keys along with a knob for that one. 

Once anything is in a printable state, I'll make the files available to freely download.

![MiniDexed TQ2](https://user-images.githubusercontent.com/5728860/165201204-737bf01a-011c-4629-86d2-9e7f311312e6.png)

**Reply by JoaquinSainz** [reply]
Wow this is great, which model of raspberry pi are you planning to use for this case?

**Reply by probonopd** [reply]
[Raspberry Pi Zero 2 W](https://www.raspberrypi.com/products/raspberry-pi-zero-2-w/) without headers would fit nicely :)

I happen to have one coming in tomorrow.

**Reply by now-its-dark** [reply]
> Wow this is great, which model of raspberry pi are you planning to use for this case?

I'm designing it around a pi3b initially, but it's a "floating" style topcase, so different bottom cases (not visible in the render) could be produced fairly easily, to accommodate various models / DAC configurations. Planning to get the internal dimensions dialed-in to be as versatile as possible. 😁

**Comment by romix123** [comment]
Had missed this thread.
Hi all, I have modified a case design for a Monome Norns Shield to hold the miniDexed hardware. It is V1, as it requires glue to hold encoder and DAC, but for those eager to put their kit in a box., here you go.
[https://www.thingiverse.com/thing:5371115](https://github.com/probonopd/MiniDexed/discussions/url)
![904E5A9F-D23F-44E0-8CD1-9619EE320311](https://user-images.githubusercontent.com/1766442/165786407-e4d6f678-b36f-4cb5-8e05-c3a32d7749b4.jpeg)
![CAB5E479-9EF4-4DA1-BC50-8198AB807329](https://user-images.githubusercontent.com/1766442/165786432-bb594c9e-46b1-4823-834f-8f74b83b8419.jpeg)

**Reply by probonopd** [reply]
Hello @romix123, thanks for sharing.

**Reply by paulsijben** [reply]
I built it over the weekend.
Couple of tips for improvement;

1. The hole for the LCD was too small for my display. Perhaps it would help if the spacers for the display we higher so the display can live completely inside the case
2. and the dupont cables I used for the connection to the PI were rather stiff so the case did not want to stay closed afterwards. So i just keep it closed with tape for now

**Comment by romix123** [comment]
It will accommodate all Pi’s, I think. 2 and 3 in any case.

Verstuurd vanaf mijn iPad

**Comment by rcarmo** [comment]
So here's my draft OpenSCAD design, based on a Pi 3A+:

![image](https://user-images.githubusercontent.com/392683/166745371-17d872e6-fa0c-4b21-86ca-142d90f27891.png)

I'm trying to work out things so that the encoder knob does not protrude (think that's good for now, need to reason out the actual knob after I finish the top case) and am waiting for a DAC to arrive so I can shimmy it under the display..

**Reply by rcarmo** [reply]
First test print - all ports are mostly correct, the DAC has a snap fit and other than some first layer printing issues (which is why the front looks messy), the case snaps together and allows for SD card access just fine:

<img width="819" alt="image" src="https://user-images.githubusercontent.com/392683/168495001-b4e68d94-55d0-4b5d-8d66-7ee056a1d913.png">
<img width="514" alt="image" src="https://user-images.githubusercontent.com/392683/168495072-39a03784-d332-40cc-9af5-875f35e8070f.png">

**Reply by probonopd** [reply]
Wow, very nice. To do this in OpenSCAD must be a lot of work!

**Reply by rcarmo** [reply]
It's kind of fun, really. I've been meaning to do new enclosures in Blender but I like the precision of OpenSCAD.

**Reply by zolkin** [reply]
@rcarmo do you mind sharing STLs?

**Comment by gstrug** [comment]
I built this one based on a Pi 4 I had lying around. Put the case on thingyverse https://www.thingiverse.com/thing:5387567. I am not a 3d designer so case may need some tweaking but good enough for my purposes
![minidexed_s](https://user-images.githubusercontent.com/98769097/168778776-4f99fa90-39c9-49a4-81ed-5c6f34d238c1.jpg)
![minidexed1](https://user-images.githubusercontent.com/98769097/168779491-18659f3d-7205-4371-b7da-91ca84e4e4da.png)
.

**Reply by craigyjp** [reply]
It has a TX7 look about it.

**Reply by pik33** [reply]
I "remixed" this one to fit HifiBerry Light. I had to add holes for HifiBerry outputs, but I didn't manage to mount all these parts. HifiBerry collided with the LCD.

So I made a bigger version of this case and this is the result:
![1665852072925](https://user-images.githubusercontent.com/19744270/195998243-b6d5cbf0-3a7d-42ae-be26-355fc16d6c8d.jpg)

**Comment by crazyhendrix** [comment]
![IMG_20220518_195313.jpg](https://user-images.githubusercontent.com/7188927/169032802-51695102-e747-4c12-bff3-dd527fb1b819.jpg)

Not a 3D printed case, but I hacked together some plastic boxes I had into a case. Yeah, it looks bad and diy, but it works and I love the look. :D

**Comment by probonopd** [comment]
![soc-minidexed](https://user-images.githubusercontent.com/2480569/169104783-d66f1182-0af2-4d71-b7d1-9f9ad3717dc2.jpg)

![minidexed](https://user-images.githubusercontent.com/2480569/169104859-4e5d0608-c535-4967-a8cb-b2915f0fe847.jpg)

![caseinside](https://user-images.githubusercontent.com/2480569/169105081-a27e8bb3-65fb-43cf-a4f7-2a30d5c5b6bf.jpg)

But the best part is when everything just snaps together nicely, no screws used.

Which means that it can be assembled in seconds with no tools (switch on the sound to hear those satisfying clicks when it all snaps together):

https://user-images.githubusercontent.com/2480569/169106573-4f663e35-6cc3-41ec-9b06-deb62a46ded5.mp4

And is tinker-friedly because it is easy and fast to take apart (switch on the sound to hear those satisfying noises):

https://user-images.githubusercontent.com/2480569/169106148-19373834-dceb-4a5b-be38-b15a24dfc817.mp4

The case is designed for:
* Raspberry Pi 2, 3, or 4 Model B (they have different places for the connectors but it is designed so that all of them fit). Actually Model B is overkill for MiniDexed but it is all I had at hand and maybe someone wants to connect more than one USB MIDI controller and/or will do something with the Ethernet port some day
* GY PCM5102 DAC module
* Rotary encoder with nut
* Display module with i2c "backpack"

Please find the file below. I am not sure whether the dimensions of all those rotary encoders and 16x2 display modules are similar enough, so feedback is welcome!

[RPi_2_3_4.zip](https://github.com/probonopd/MiniDexed/files/9184430/RPi_2_3_4.zip)

**Reply by JoaquinSainz** [reply]
Wow that's really cool!! The idea that is assembled quickly is very nice. Seeing the LCD that you are using and the encoder, it seems that they have the same measurements as mine.  I am really interested in those files.

**Reply by rcarmo** [reply]
The encoder breakout (HW-040) and LCD should be the same dimensions - they're pretty much standard parts.

I like the angle, have been thinking about incorporating something like that into my design so that I can use the extra height in the back to fit in a TRS MIDI jack...

**Reply by JoaquinSainz** [reply]
Hey how you doing @probonopd, Currently I do not have any knowledge on how to make 3d designs, I am about to take online courses to learn and contribute to this project... Do you think you could share the files with me so that I have a base and make some models starting from that? I'm thinking of adding some tr 6.35mm audio outputs.
Thanks

**Reply by dhstsw** [reply]
That's great, i've been waiting for this.
Could you also post the STEP files for it? I have to accomodate inside also a potentiometer and a Raspberry Pico in it :)
THX

**Comment by romix123** [comment]
@JoaquinSainz, tip, explore Tinkercad (free online). Relatively easy app to create 3d designs using addition and subtraction with basic shapes.

**Reply by probonopd** [reply]
Highly recommend watching some videos about how to make [snap fit cases](https://www.youtube.com/results?search_query=snap+fit+case). Can be done in TinkerCad, but other applications are way more powerful.

**Reply by JoaquinSainz** [reply]
Thanks @probonopd  what other app do you recommend?

**Reply by probonopd** [reply]
I recommend you to watch the linked videos, understand the _concepts_ and then choose any tool that makes sense to you. Many examples are based on Fusion 360 but that's the one I do NOT recommend because it has complicated licensing, phones home, and is not available for Linux. Check out the alternatives, including Onshape.

**Comment by chemati8** [comment]
Could you upload the files to print in 3d?. Thanks

**Reply by JoaquinSainz** [reply]
It would be really nice if they share the designs that everybody has made. Please share files🙏🙏🙏

**Reply by romix123** [reply]
See 
https://www.tinkercad.com/things/1lFmZdBiwyV-minidexed 
https://www.thingiverse.com/thing:5371115

**Comment by dhstsw** [comment]
Hi all,
This is my take on a printable case for Minidexed.

This is for Rpi3 / Rpi2 but i guess with minor mods may fit an Rpi4 too (USB and ETH port fittings).

I needed one case for the project, but with the ability to fit in also a Raspberry Pico (for midi-host compatibility with DAWs), a stereo RCA out (i had one around, why not to use it?) and a potentiometer for the display (i don't have a i2s display, so this is for a W 202 potentiometer).

The W202, PCM5102 DAC and Raspberry Pico are snapping in place without the need of screws (you can use up to 4 M2 screws on the Pico if you want, holes are there, but really not needed).
![IMG_20220829_110103](https://user-images.githubusercontent.com/59790194/187203314-85429cc7-8520-41fe-8a44-ef12ea115d7d.jpg)

Model has been printed and adjusted for tolerances and is pretty spot on on my printer.
No need to compensate for xy expansion (i already ajusted tolerances on the models).

BTW: you can connect the VSYS (pin 39) and GND (pin 38) to the 5V and GND on the Rpi, so you'll just need one cable (connected to the Pico) to have both midi and energy for the whole system.

Also, it's the first time i ever use a parametric CAD in my life, so be kind  :)

Hope it will be of use for someone beside me :)
![1](https://user-images.githubusercontent.com/59790194/187204200-4247d5f4-32ea-465b-9e7e-eb94abe87f68.jpg)
![IMG_20220829_153043](https://user-images.githubusercontent.com/59790194/187204213-e4c287c3-3d96-4204-82a2-4dce75a1ec41.jpg)

[Minidexed_DHS.zip](https://github.com/probonopd/MiniDexed/files/9444588/Minidexed_DHS.zip)

**Reply by zsoltmar** [reply]
Nice one! Can you please share the Pico project for the midi host?

**Comment by rcarmo** [comment]
I like the angled display - I’ve been thinking of redesigning mine to have something similar (and make room for a TRS MIDI breakout board)

**Comment by ghost** [comment]
Nicely integrated with my modular rig, using a 3d printed 19 inch rack blanking plate. 
![16695813418963812677629251492301](https://user-images.githubusercontent.com/103966073/204158411-424d6ed5-e3c5-4591-9736-435979056317.jpg)

**Reply by paulsijben** [reply]
nice!

**Reply by craigyjp** [reply]
A bit like my 19" rack

![20220422_204118.jpg](https://user-images.githubusercontent.com/51026305/204244797-3e343a2b-7d65-43a3-bf65-c0f0cdf9a029.jpg)

**Reply by zsoltmar** [reply]
Are you controlling it with CV??

**Reply by ghost** [reply]
Hi Zsoltmar, no - midi over usb. Works well with no latency issues.

**Comment by zsoltmar** [comment]
![minidexed](https://user-images.githubusercontent.com/9203999/210873356-292c8714-9f38-404f-a84b-91d5b668712f.jpg)
Here's mine if anyone wants to build one: https://www.printables.com/model/359937-minidexed-case-w-rpi-3-batteries-switches-screen/files

**Comment by b1tfl0w** [comment]
Hello, I share my case also. 
I had to increase the height by 5 millimeters in the model I shared, because  the panel was too close to the raspberry pi pins. 
I'm not sure if the audio card fits in the slot because I've placed in another way inside since I'm only using a big stereo jack.
The other hole is for 12 v power jack, inside I placed a dc to dc converter (lm2596) set to 5.1 v.
It would also need more screws maybe but I'm quite happy for now!
[https://www.thingiverse.com/thing:6965282](https://www.thingiverse.com/thing:6965282)
![IMG_20250227_220150](https://github.com/user-attachments/assets/68d66e62-b0fe-4851-9c1c-2a56bd42422c)
![IMG_20250227_220211](https://github.com/user-attachments/assets/83a4d418-5928-45ef-8200-95f4bebd433a)
![IMG_20250227_220323](https://github.com/user-attachments/assets/3bb2eb81-996c-4f64-9151-dd253d99c9a4)
![IMG_20250227_220404](https://github.com/user-attachments/assets/03d3b18a-8bbf-416e-832d-732b09ddb6dc)
